lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,117 @@
+import glob
+from pathlib import Path
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import Unpack
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tar_name = "A.tar.xz"
+    tarlist = [tar_name]
+    linksrc = "TO-UNPACK"
+
+    def mock_glob(self, path):
+        """Returns the Tarball list"""
+        return TestUnpackTarballs.tarlist
+
+    def mock_resolve(self: Path, strict: bool = False):
+        """
+        Return a "full" fake file name
+
+        Args:
+            strict: Not used in the mock
+
+        Returns:
+            new Path with fake resolution
+        """
+        return Path(self.tar_name)
+
+    def test_tarball_symlinks(self, monkeypatch, server_config, make_logger, caplog):
+        """Tests for an incorrect symlink which results in an unresolved location"""
+
+        monkeypatch.setattr(glob, "iglob", self.mock_glob)
+        obj = Unpack(server_config, self.linksrc, make_logger)
+        total, success = obj.unpack_tarball()
+        assert (
+            f"Tarball link, '{self.tar_name}', does not resolve to a real location"
+            in caplog.text
+        )
+
+    def test_tarball_file(self, monkeypatch, server_config, make_logger, caplog):
+        """Show that when tarball path cannot be accessed it throws a
+        FileNotFoundError Exception"""
+
+        expected_error_msg = f"[Errno 2] No such file or directory: '{self.tar_name}'"
+        monkeypatch.setattr(glob, "iglob", self.mock_glob)
+        monkeypatch.setattr(Path, "resolve", self.mock_resolve)
+        obj = Unpack(server_config, self.linksrc, make_logger)
+        with pytest.raises(FileNotFoundError) as exc:
+            total, success = obj.unpack_tarball()
```

Comment:
I believe that the exception here is being raised by `get_tarball_md5()`'s attempt to access the file system.  

Now, not to be repetitive or anything, but....
- we should probably have a `try` block wrapped around `get_tarball_md5()` which catches this error;
- we should be mocking our external dependencies, particularly the file system; and,
- if we want a particular behavior from `get_tarball_md5()`, we should mock it.